doc/source/ray-air/examples/pytorch_resnet_finetune.ipynb

```diff
@@ -0,0 +1,917 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 19,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from ray.train.torch import TorchCheckpoint\n",
+    "\n",
+    "\n",
+    "def initialize_model():\n",
+    "    # Load pretrained model params\n",
+    "    model = models.resnet18(pretrained=True)\n",
+    "\n",
+    "    # Replace the original classifier with a new Linear layer\n",
+    "    num_features = model.fc.in_features\n",
+    "    model.fc = nn.Linear(num_features, 2)\n",
+    "\n",
+    "    # Ensure all params get updated during fintuning\n",
+    "    for param in model.parameters():\n",
+    "        param.requires_grad = True\n",
+    "    return model\n",
+    "\n",
+    "\n",
+    "# You can also initialize a model from previous checkpoint\n",
+    "CHECKPOINT_URI = \"s3://air-example-data/finetune-resnet-checkpoint/TorchTrainer_4f69f_00000_0_2023-02-14_14-04-09/checkpoint_000000/\"\n",
+    "\n",
+    "\n",
+    "def initialize_model_from_ckpt():\n",
+    "    checkpoint = TorchCheckpoint.from_uri(CHECKPOINT_URI)\n",
+    "    resnet18 = initialize_model()\n",
+    "    return checkpoint.get_model(model=resnet18)\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Define the Training Loop\n",
+    "\n",
+    "The `train_loop_per_worker` function defines the finetuning procedure for each worker.\n",
+    "\n",
+    "**1. Load dataset shard for each worker**:\n",
+    "- A ray trainer will take a dictionary of ray datasets as input. One can accessed these data by `session.get_dataset_shard(DATASET_KEY)` in the workers.\n",
+    "- Only the dataset with key \"train\" will be split into multiple shards, while all the others will remain the same. \n",
+    "- One can use {meth}`iter_torch_batches <ray.data.Dataset.iter_torch_batches>` to iterate the datasets with automatic tensor batching. If you need more flexible customized batching function, please refer to our lower-level api {meth}`iter_batches <ray.data.Dataset.iter_batches>`.\n",
+    "\n",
+    "**2. Prepare your model**:\n",
+    "- `train.torch.prepare_model` will prepares the model for distributed training. Under the hood, it converts your torch model to `DistributedDataParallel` model, and synchronizes the gradients and buffers across all workers.\n",
+    "\n",
+    "**3. Report metrics and checkpoint**:\n",
+    "- `session.report` will gather the metrics from each worker and save them into log files.\n",
+    "- You don't have to save checkpoints manually with `torch.save()`, `session.report()` will help you sync checkpoints to local/cloud storage.\n",
+    "- The best checkpoints will be saved according to the specified `checkpoint_score_attribute` in {class}`CheckpointConfig <ray.air.config.CheckpointConfig>`. Here we only save the best model with highest validation accuracy."
```

Comment:
Update this. We're now saving latest checkpoint. We should also move this to where we define the `CheckpointConfig` in the code.